src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php

```diff
@@ -227,4 +227,44 @@ public function testSwitchUserKeepsOtherQueryStringParameters()
         $this->assertSame('page=3&section=2', $this->request->server->get('QUERY_STRING'));
         $this->assertInstanceOf('Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken', $this->tokenStorage->getToken());
     }
+
+    public function testSwitchUserWithReplacedToken()
+    {
+        $token = new UsernamePasswordToken('username', '', 'key', array('ROLE_FOO'));
```

Comment:
`key` should be `provider123` (or the other way around) to be consistent with the `SwitchUserListener` definition below